src/python/zquantum/core/graph.py

```diff
@@ -73,163 +66,223 @@ def generate_graph_node_dict(graph: nx.Graph) -> dict:
+    """(Deprecated) select graph generator by its name and arguments in the graph specs.
+
+    This is deprecated because each graph generator function requires slightly different
+    arguments, so wrapping all of them is convoluted and results in deeply nested dicts.
+
+    Graph generation is controlled via the "type_graph" key. Each graph generator
+    requires additional arguments passed as additional keys in the `graph_specs` dict.
+    The available `type_graph` values are:
+        - "erdos_renyi". Additional keys:
+            - "probability" (int)
+        - "regular". Additional keys:
+            - "degree" (int)
+        - "complete". Same as "erdos_renyi" with probability set to 1.0.
+        - "caveman". Additional keys:
+            - "number_of_cliques" (int)
+            - "size_of_cliques" (int)
+        - "ladder". Additional keys:
+            - "length_of_ladder" (int)
+        - "barbell". Additional keys:
+            - "number_of_vertices_complete_graph" (int)
 
     Args:
-        graph_specs: dictionary
-            Specifications of the graph to generate. It should contain at
-            least an entry with key 'type' and one with num_nodes
+        graph_specs: specification of graphs to generate. Required keys:
+                - "type_graph" (str)
+                - "num_nodes" (int)
+                - "random_weights" (bool) - if True, weights are sampled from U(0, 1).
+                    Defaults to False.
+                - additional arguments to graph generator depending on the choice of
+                    "type_graph" (see above)
+            Optional keys:
+                - "seed"
 
-    Returns:
-        A networkx.Graph object
     """
-    type_graph = cast(str, graph_specs["type_graph"])
-    num_nodes = cast(int, graph_specs.get("num_nodes"))
-    random_weights = graph_specs.get("random_weights", False)
-    seed = graph_specs.get("seed")
-    number_of_cliques = cast(int, graph_specs.get("number_of_cliques"))
-    size_of_cliques = cast(int, graph_specs.get("size_of_cliques"))
-    length_of_ladder = cast(int, graph_specs.get("length_of_ladder"))
-    number_of_vertices_complete_graph = cast(
-        int, graph_specs.get("number_of_vertices_complete_graph")
+    warnings.warn(
+        "zquantum.core.generate_graph_from_specs is deprecated. Please use other, "
+        "specialized functions from this module directly.",
+        DeprecationWarning,
     )
 
+    type_graph = graph_specs["type_graph"]
+    num_nodes = graph_specs["num_nodes"]
+    if graph_specs.get("random_weights", False):
+        weight_sampler = uniform_sampler(0, 1)
+    else:
+        weight_sampler = static_sampler()
+
+    seed = graph_specs.get("seed")
+
     if type_graph == "erdos_renyi":
-        probability = graph_specs["probability"]
         graph = generate_random_graph_erdos_renyi(
-            num_nodes, probability, random_weights, seed
+            num_nodes,
+            graph_specs["probability"],
```

Comment:
Just double checking - should this be "edge_probability"?

Reply:
yeah, makes more sense to me to, but I'd keep it for backwards compatibility. The whole function will be deprecated anyway